### PR TITLE
[DART] Add standard generated classes as reserved

### PR DESCRIPTION
**Pull request by sbu-WBT**
A friend create a specification with a model named `ApiClient` which resulted for me in two classes with the same name which doesn't compile. So I thought it might be handy to mark the default included classes also as reserved.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@swipesight (2018/09) @jaumard (2018/09) @josh-burton (2019/12) @amondnet (2019/12) @sbu-WBT (2020/12) @kuhnroyal (2020/12) @agilob (2020/12)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java
Comment (kuhnroyal) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java:168]: This should also go to the normal dart client as it is part of the `http` package and not used in the dio generator?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartClientCodegen.java
Comment (agilob) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartClientCodegen.java:42]: Thinking about the opposite problem. I don't generate ApiClient class, I would generate only models. This makes ApiClient a non-reserved word, thus produces unexpected results.

The problem exists, but I'm not convinced this is the right solution here.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java
Comment (agilob) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java:168]: This change is in AbstractDartCodegen.java so it affects all dart clients, including dartdio.

Comment (kuhnroyal) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java:168]: Yea so it should also go to `DartClientCodegen`.

Comment (sbu-WBT) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java:168]: I thought of this,  too. But then I saw [`Response`](https://pub.dev/documentation/http/latest/http/Response-class.html) which is also a `http` class (If I'm not wrong) so I thought I put it there. I have no problem to put it in the other set

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartClientCodegen.java
Comment (kuhnroyal) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartClientCodegen.java:42]: I think this is the correct solution, doing this as well in the dio generator. You may have a couple false positives but that is ok I'd say.

Comment (sbu-WBT) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartClientCodegen.java:42]: There is the option to only generate the models? Then just add the classes to the set only if the parameter is not set. Do you mind telling me the option name?

Comment (agilob) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartClientCodegen.java:42]: >Or if you have another idea just tell us, I'm open to better solutions

Well, no. It would have to be conditionally added increasing complexity of such solution. I would prefer to see global PR for this problem rather than each generator doing it on their own.

@wing328 is there any DefaultCodeGen solution for this? A set of _potentially_ used keywords like `ApiClient` to prevent generated files collide with project files. It would have to be managed independently by each code generator and their cmd options.

Looks like big effort to handle some small cases, but correctness of the output should be priority(?).

**Comment by agilob** [comment]
Im happy to approve it after @kuhnroyal approves it. I think the problem in other comments needs a wider planning for all generators, making it out of scope for your problem @sbu-WBT

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java
Comment (kuhnroyal) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java:168]: Yea, I added `Response` here because it also exists in Dio (a different class but same name). Same goes for `Client`.
I think it doesn't hurt here as the `http` package can basically be considered part of Dart.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartClientCodegen.java
Comment (kuhnroyal) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartClientCodegen.java:42]: > is there any DefaultCodeGen solution for this? A set of _potentially_ used keywords like `ApiClient` to prevent generated files collide with project files. It would have to be managed independently by each code generator and their cmd options.

The problem is that most other generators/languages have a concept of absolute/fq imports but Dart does not.
So in other generators you can use the import mapping which can be configured as cmd parameters or in the generator class. The dart generators basically miss uses the import mapping, meaning we HAVE TO use it differently due to the absence of the absolute/fq imports concept.

The `additionalReservedWords ` is just something I came up with for the dio generator to solve this in a simple manner.

Comment (kuhnroyal) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartClientCodegen.java:42]: Well that is also not fully correct. Dart has named imports `import "package:foo:foo.dart" as bar;` but I tried this and it fails with a lot of builders/build_runner (`json_serializable`, `built_value`).

**Review by kuhnroyal — APPROVED** [review]

**Review by agilob — APPROVED** [review]

**Comment by agilob** [comment]
@sbu-WBT re-run the tests

**Comment by wing328** [comment]
@sbu-WBT thanks for the PR, which has been included in the v5.1.0 release: https://twitter.com/oas_generator/status/1373636654024380423

**Pull request merged**
